### PR TITLE
Loop until the mouse is grabbed

### PR DESCRIPTION
**Pull request by omixann**
Right now, if the mouse cannot be grabbed, crub continues as if nothing happened, the window is created and stays there until you kill the process. With this change, crub will try to grab the mouse in an infinite loop, but without hogging the CPU.

### CODE REVIEW

**Comment by omixann** [comment]
I'm thinking that maybe it's not the best idea to keep crub (which might be invisible to the user) waiting until the cursor is ungrabbed by default. Would it be better if hidden behind a command line parameter?

File: crud.c [hunk 1]
```diff
@@ -80,10 +80,14 @@ void set_rects_from_selection(XRectangle* rects, Selection dimensions) {
...
-  XGrabPointer(display, root, True,
-               mask, GrabModeAsync, GrabModeAsync,
-               None, *cursor, CurrentTime);
+  int error;
+
+  do {
+    error = XGrabPointer(display, root, True,
+                 mask, GrabModeAsync, GrabModeAsync,
+                 None, *cursor, CurrentTime);
+    sleep(0);
```
Comment (ix) [inline crud.c:89]: i haven't had time to actually try this out yet, but is there a justification for sleeping here?

**Comment by ix** [comment]
maybe it's better to just exit out if the cursor can't be grabbed? it sounds like a reasonable cause for failure, and the user can always go ahead and rerun when the cursor is freed up. i don't like the idea of the program waiting an arbitrary length of time to grab it.

**Comment by jmi2k** [comment]
I'll tell you my usecase: I've bound a right-click on the root window to a script which uses crud. The problem is that it triggers when the user presses the button, so it is grabbed when crud is started. This waits until the click is released, so my binding works.

I agree, the default behavior should be exiting (in fact, I'm having the same problem with xkill, which I also want to bind). But do you see this behavior useful enough to keep it behind a command line flag or a config option?

**Comment by omixann** [comment]
_I accidentally deleted my comment, so I'll recreate it_

I'll tell you my usecase: I have a mouse binding which triggers when I click on the root window. When the button is pressed, a script which contains crud is started. The problem is that when that script runs, the button is still pressed, so the mouse is grabbed and the script fails.
I agree, looping forever it's not a good idea, exiting seems like the way to go (like, for example, xkill). But now I wonder, is that behavior useful enough to be kept behind a command line parameter or a config flag?

**Comment by ix** [comment]
a config flag is a fine idea, adding option parsing is something i'm trying to avoid for its verbosity.
perhaps `PERSISTENT_GRABBING` ?

**Comment by omixann** [comment]
Done! I think it's only missing a bit of cleanup instead of just `exit(1)` though but you get the idea

**Pull request merged**